### PR TITLE
Ian_McNicholas Take-away Challenge Week 2

### PR DESCRIPTION
**Pull request by ianmcnicholas**
I haven't gone very far with edge cases.  e.g. there is no way for a customer to remove an item from the order if they make a mistake.
The text function is working.
I really struggled with the testing of this project.  My test coverage is only 80%.  Two main issues were if a method 'puts'es a statement e.g. viewing a menu, I am unaware how to test for this, and also I could not think how to test any of the text.rb file - this includes whether or not to use a double or the Time.now function.
I have however used a double as required in the order_spec.rb file.
My main concern is that I have not grasped the learning objectives from week 2, and this is evidenced in this project.

### CODE REVIEW

File: lib/menu.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+                 { :pizza => 'mediterranean', :price => 8 },
+                 { :pizza => 'pepperoni', :price => 9 },
+                 { :pizza => 'four-cheese', :price => 10 },
+                 { :pizza => 'meat-feast', :price => 11 }]
+  end
+
+  def view
+    puts "PIZZA".ljust(20) + "PRICE"
+    @contents.each { |item| puts "#{item[:pizza]}".ljust(20) + "£#{item[:price]}" }
+  end
```
Comment (kerisic) [inline lib/menu.rb:16]: Great formatting for the display of the menu here!

File: lib/order.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  def initialize
+    @basket = []
+    @menu = Menu.new
+  end
+
+  def add(item, quantity = 1)
+    fail "That is not on the menu" unless @menu.contents.any? { |hash| hash[:pizza] == item }
+
+    quantity.times { @menu.contents.each { |option| @basket << option if option[:pizza] == item } }
+  end
```
Comment (kerisic) [inline lib/order.rb:17]: Like the neatness of the method for adding into basket.

File: spec/text_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require './lib/text.rb'
+
+describe Text do
+
+  # it "initializes with the customer's phone number" do
+  #   text = Text.new("phone_number")
+  #   expect(text).to respond_to(:phone_number)
+  # end
```
Comment (kerisic) [inline spec/text_spec.rb:8]: Have a look at stubbing texting functionality guide here [https://github.com/makersacademy/course/blob/master/pills/levels_of_stubbing.md]

**Review by kerisic — COMMENTED** [review]
The methods are neat and easy to understand with elegant blocks for the hashes.

File: spec/menu_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    it "has a contents(menu) with prices on it" do
+      expect(subject.contents.first[:price]).not_to eq nil
+    end
+
+  end
+
+  it "allows the customer to view the menu" do
+    expect(subject).to respond_to(:view)
```
Comment (tatsiana-makers) [inline spec/menu_spec.rb:22]: We can use the `output` matcher here to test printing: https://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#new-output-matcher

**Review by tatsiana-makers — COMMENTED** [review]
This is a nice go at the challenge! Thanks for asking for feedback Ian (: Let me know if you have questions about any of my comments
